jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -133,9 +134,51 @@ public void assignGroupToRole(final String serverTemplateId, final String contai
     }
 
     @Override
-    public void assignGroupAndUserToRole(final String serverTemplateId, final String containerId, final String caseId, final String roleName, final String user, final String group) {
-        client.assignGroupToRole(containerId, caseId, roleName, group);
-        client.assignUserToRole(containerId, caseId, roleName, user);
+    public void assignToRole(final String serverTemplateId, final String containerId, final String caseId,
+                             final String roleName, final List<String> usersList, final List<String> groupsList) {
+        CaseRoleAssignmentSummary ras = getCaseInstance(serverTemplateId, containerId, caseId).getRoleAssignments()
+                .stream()
+                .filter(ra -> roleName.equals(ra.getName()))
+                .findFirst().orElse(null);
+
+        if (ras != null) {
+            if (ras.getUsers() != null) {
+                List<String> usersToremove = new ArrayList();
+                ras.getUsers().forEach(
+                        user -> {
+                            if (!usersList.stream().filter(newU -> user.equals(newU)).findFirst().isPresent()){
+                                usersToremove.add(user);
```

Comment:
Instead of creating a new "holder" list for the users which are going to be removed, maybe it makes sense to call immediately:
_removeUserFromRole(serverTemplateId, containerId, caseId, roleName, user)_
That way, you wouldn't need to create a separate iteration for this.